front/src/components/CTF/Calendar.vue

```diff
@@ -96,13 +134,30 @@ function dateToLocale(s: string, offset = 0): string {
 export default defineComponent({
   components: { QCalendarMonth },
   setup() {
+    const icalEl = ref<QInput>();
     const { result: ctfs, loading } = ctfnote.ctfs.getAllCtfs();
+    const { result: icalPassword } = ctfnote.settings.getIcalPassword();
+
+    const icalLink = computed(() => {
+      icalEl.value?.select();
+      if(icalPassword.value != ''){
+        return (
+          document.location.origin +
+          '/calendar.ics?key=' +
+          encodeURIComponent(icalPassword.value)
+        );
+      }
+      return document.location.origin + "/calendar.ics";
```

Comment:
This has a similar problem to my other comment.

Maybe I should instead read the entire document.location and replace everything after '#' with /calendar.ics? Again, feedback is appreciated.